Resolves extra padding on recording cell top edge

samgrover opened the pull request:
<!-- Thanks for contributing to Science Journal iOS! Before you submit your pull request, please make sure to check the following boxes by putting an x in the [ ] (don't: [x ], [ x], do: [x]) -->

### Checklist
- [x] All new and existing tests pass
- [x] I've read the [Contribution Guidelines](https://github.com/google/science-journal-ios/blob/master/CONTRIBUTING.md)
- [x] I've read [Change Limitations](https://github.com/google/science-journal-ios/blob/master/CHANGE_LIMITATIONS.md)

### Motivation and Context
<!-- Why is this change required? What problem does it solve? -->
<!-- If it fixes an open issue, please link to the issue here. -->
https://github.com/google/science-journal-ios/issues/46

### Description
<!-- Describe your changes in detail -->
Provide zero section insets when there are no cells in the corresponding section.
<!-- Please describe in detail how you tested your changes. -->
Manual testing of various UI states in multiple orientations, on iPhone and iPad simulators.

samgrover commented:
I signed it!

garrettmurray commented on ScienceJournal/UI/ExperimentItemsViewController.swift:
```diff
@@ -436,9 +436,7 @@ class ExperimentItemsViewController: VisibilityTrackingViewController,
-    if experimentDataSource.isArchivedFlagSection(section) &&
-        experimentDataSource.numberOfItemsInSection(
-        experimentDataSource.archivedFlagSectionIndex) == 0 {
+    if experimentDataSource.numberOfItemsInSection(section) == 0 {
```
So as discussed, the one issue with this is that, depending on the sections visible, you can still end up with a case where a section has a bottom padding and the next section has a top padding, which causes a double-padding scenario (such as having items in an experiment then starting a recording). Here's a proposed solution that _seems_ to solve this in all cases:

```
    // Determine the last section with items.
    var lastSectionWithItems:Int? = nil
    for i in 0..<collectionView.numberOfSections {
      if collectionView.numberOfItems(inSection: i) > 0 {
        lastSectionWithItems = i
      }
    }

    if experimentDataSource.numberOfItemsInSection(section) == 0 {
      // If the section is empty, no insets.
      return .zero
    } else if section == lastSectionWithItems {
      // If this is the last section, include the bottom inset.
      return cellInsets
    } else {
      // If this is not the bottom section, but this section has items, use standard insets without
      // bottom. This avoids an issue where two stacked sections
      var modifiedInsets = cellInsets
      modifiedInsets.bottom = 0
      return modifiedInsets
    }
```

samgrover commented:
Updated using your code, with a minor change to use data source instead of collection view in the first block.

Also, removed the outdated comments on the method as per our discussion.
